src/nvim/screen.c

```diff
@@ -3805,8 +3804,7 @@ win_line (
           && ((area_attr != 0 && vcol == fromcol
                && (VIsual_mode != Ctrl_V
                    || lnum == VIsual.lnum
-                   || lnum == curwin->w_cursor.lnum)
-               && c == NUL)
```

Comment:
c == NUL is always true. See line 3780

Reply:
I would usually add `assert(c != NUL);` just to be safe. In this case it's pretty obvious that the entire block is predicated on that fact, but even then ...